### ISSUE TITLE
[IDEA] Small system macro/variable to support coding (full solution included)

### ISSUE THREAD

**Issue by AnthonyMuscio** [Jermolene/TiddlyWiki5#4729]
**Is your feature request related to a problem? Please describe.**
A Tiddler transcluded into the current tiddler, currently Has no access to its own tiddlername 
eg `{{||tiddlername}}` This means that references to tiddlername!!filename needs to be hard coded in tiddlername
to access its own fieldname.

**Describe the solution you'd like**
Place the following macros into the core or a version there of
```
\define pipe() |
\define codeTiddler() {{{ [<transclusion>split<pipe>nth[2]] }}}
```

**Describe alternatives you've considered**
Many with no luck, macro distribution is fine but it would be helpful to all and not need a conscious install step it it were in the core. It is a general solution.

**Additional context**
The following code shows how this variable can be used in code tiddlers that are transcluded
```
<$wikify name=code-tiddler text="""<<codeTiddler>>""">
<$list filter="[all[current]match<code-tiddler>]">
  In Code tiddler <$view field=fieldname/>
</$list>
<$list filter="[all[current]!match<code-tiddler>]"variable=nul>
  Transcluded In Calling tiddler <$view field=fieldname/>
</$list>
</$wikify>
```
Note that the code tiddler can contain code that displays only when its open in the story, and access its own fields.

**A Better solution?**
If the codeTiddler macro could return a value that did not need wikify before use it would be much easier.

**Comment by AnthonyMuscio** [comment]
Post script, I have developed a new concept called autonomous fields and tags, and hope to take this even further

[See here](https://groups.google.com/forum/?hl=en#!topic/tiddlywiki/RP3RYc6Opxc)

The ability to determine the name of the current codeTiddler will be invaluable by permitting a single tiddler provide information when viewed that does not interfere when it is transcluded, in view templates or toolbars. Further building on its autonomous nature.

Regards
Tony

**Comment by AnthonyMuscio** [comment]
I started a more public discussion on this [here](https://groups.google.com/forum/?hl=en#!topic/tiddlywiki/52AB4uY-6WY)

**Comment by AnthonyMuscio** [comment]
Closed as satisfied in TW 5.3.0 with the thisTiddler variable as also asked for in a recently closed issue #4216
